src/UKHO.ERPFacade.CleanUp.WebJob/Services/CleanUpService.cs

```diff
@@ -1,50 +1,58 @@
-﻿using Microsoft.Extensions.Options;
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using UKHO.ERPFacade.Common.Configuration;
-using UKHO.ERPFacade.Common.Constants;
+using UKHO.ERPFacade.Common.Enums;
+using UKHO.ERPFacade.Common.Logging;
 using UKHO.ERPFacade.Common.Operations.IO.Azure;
 
 namespace UKHO.ERPFacade.CleanUp.WebJob.Services
 {
     public class CleanUpService : ICleanUpService
     {
+        private readonly ILogger<CleanUpService> _logger;
         private readonly IOptions<CleanupWebJobConfiguration> _cleanupWebjobConfig;
         private readonly IAzureTableReaderWriter _azureTableReaderWriter;
         private readonly IAzureBlobReaderWriter _azureBlobReaderWriter;
 
-        public CleanUpService(IOptions<CleanupWebJobConfiguration> cleanupWebjobConfig,
-                               IAzureTableReaderWriter azureTableReaderWriter,
-                               IAzureBlobReaderWriter azureBlobReaderWriter)
+        public CleanUpService(ILogger<CleanUpService> logger,
+                              IOptions<CleanupWebJobConfiguration> cleanupWebjobConfig,
+                              IAzureTableReaderWriter azureTableReaderWriter,
+                              IAzureBlobReaderWriter azureBlobReaderWriter)
         {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _cleanupWebjobConfig = cleanupWebjobConfig ?? throw new ArgumentNullException(nameof(cleanupWebjobConfig));
             _azureTableReaderWriter = azureTableReaderWriter ?? throw new ArgumentNullException(nameof(azureTableReaderWriter));
             _azureBlobReaderWriter = azureBlobReaderWriter ?? throw new ArgumentNullException(nameof(azureBlobReaderWriter));
         }
 
-        public void Clean()
+        public async Task Clean()
         {
-            CleanS57Data(PartitionKeys.S57PartitionKey);
-        }
-
-        private void CleanS57Data(string partitionKey)
-        {
-            var entities = _azureTableReaderWriter.GetAllEntities(partitionKey);
-
-            foreach (var entity in entities)
+            try
             {
-                if (entity["RequestDateTime"] == null)
-                    continue;
+                var status = new KeyValuePair<string, string>("Status", Status.Complete.ToString());
```

Comment:
Can we use Dictionary<string, string> instead of KeyValuePair

Reply:
var statusFilter= new Dictionary<string, string>
{
    { "Status", Status.Complete.ToString() }
};

Reply:
resolved